### ISSUE TITLE
Accessibility Bug: "X" button in search component is not accessible via keyboard

### ISSUE THREAD

**Issue by awinberg-aws** [aws-amplify/amplify-ui#5375]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Primitive components

### How is your app built?

CRA

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

Accessibility review has shown that the search component does not meet accessibility standards due to the clear button not being focusable via keyboard.

### What's the expected behaviour?

A user should be able to focus and activate the clear button using only their keyboard.

### Help us reproduce the bug!

https://ui.docs.amplify.aws/react/components/searchfield

in the demo, enter text into the search input. See that the clear button is displayed. Press the tab key in an effort to focus the clear button. See that the clear button is skipped and the search button is focused instead.

### Code Snippet

```typescript
// Put your code below this line.

```
